packages/flutter/test/widgets/editable_text_test.dart

```diff
@@ -554,6 +554,165 @@ void main() {
         equals('TextInputAction.done'));
   });
 
+  testWidgets('connection is closed when TextInputClient.onConnectionClosed message received', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      MediaQuery(
+        data: const MediaQueryData(devicePixelRatio: 1.0),
+        child: Directionality(
+          textDirection: TextDirection.ltr,
+          child: FocusScope(
+            node: focusScopeNode,
+            autofocus: true,
+            child: EditableText(
+              backgroundCursorColor: Colors.grey,
+              controller: controller,
+              focusNode: focusNode,
+              maxLines: 1, // Sets text keyboard implicitly.
+              style: textStyle,
+              cursorColor: cursorColor,
+            ),
+          ),
+        ),
+      ),
+    );
+
+    await tester.tap(find.byType(EditableText));
+    await tester.showKeyboard(find.byType(EditableText));
+    controller.text = 'test';
+    await tester.idle();
+
+    final EditableTextState state =
+        tester.state<EditableTextState>(find.byType(EditableText));
+    expect(tester.testTextInput.editingState['text'], equals('test'));
+    expect(state.wantKeepAlive, true);
+
+    tester.testTextInput.closeConnection();
+    await tester.idle();
+
+    // Widget does not have focus anymore.
+    expect(state.wantKeepAlive, false);
+    // testTextInput.isVisible is false if the clearClient/hideClient TextInput
+    // messages are sent. For onConnectionClosed message framework does not send
+    // clearClient/hideClient since the platform already informed the framework
+    // of a closed connection.
+    expect(tester.testTextInput.isVisible, true);
```

Comment:
Does this `expect()` add any value?

Reply:
Yes it shows that the clear or hide messages weren't called.

When clearClient/hideClient are called, it is set to false. This is the code for the test_text_input.dart:

https://github.com/flutter/flutter/blob/6bf1ad4bd5714f73c1c808dc821add7199260d0c/packages/flutter_test/lib/src/test_text_input.dart#L86

Reply:
I changed the expect to be more readable. I added a log to testTextInput